test/onnx/test_pytorch_onnx_shape_inference.py

```diff
@@ -1,72 +1,69 @@
-        input_copy = copy.deepcopy(input)
-        output = model(*input_copy)
-        if isinstance(output, torch.Tensor):
-            output = (output,)
+    def test_cast(self):
+        # Test cast with input of unknown scalar type.
+        g = self.create_empty_graph()
+        input = g.addInput()
+        cast_out = g.op("Cast", input, to_i=1)
+        self.run_test(g, cast_out.node(), expect_tensor('Float'))
 
-        _set_opset_version(self.opset_version)
-        _set_operator_export_type(OperatorExportTypes.ONNX)
-        _set_onnx_shape_inference(True)
-        _set_training_mode(False)
-        if dynamic_axes is None:
-            dynamic_axes = {}
-        _validate_dynamic_axes(dynamic_axes, model, input_names, output_names)
+    def test_constant_of_shape(self):
+        # Test ConstantOfShape with input of onnx::Shape node.
+        g = self.create_empty_graph()
+        constant = self.insert_tensor_constant(g, torch.ones(1, 2, 3, 4))
+        shape = g.op("Shape", constant)
+        constant_of_shape = g.op("ConstantOfShape", shape, value_t=torch.tensor([2.0]))
+        self.run_test(g, constant_of_shape.node(), expect_tensor('Float', (1, 2, 3, 4)))
 
-        input_copy = copy.deepcopy(input)
-        graph, _, _ = utils._model_to_graph(model, input_copy,
-                                            input_names=input_names,
-                                            output_names=output_names,
-                                            operator_export_type=OperatorExportTypes.ONNX,
-                                            example_outputs=output,
-                                            do_constant_folding=do_constant_folding,
-                                            training=TrainingMode.EVAL,
-                                            dynamic_axes=dynamic_axes)
-        verify_inferred_shape(graph)
+    def test_constant_of_shape_static(self):
+        # Test ConstantOfShape with input of prim::ListConstruct of static tensor
+        rank = 4
+        g = self.create_empty_graph()
+        constants = [self.insert_tensor_constant(g, torch.tensor(i + 1)) for i in range(rank)]
+        shape = g.op("prim::ListConstruct", *constants)
+        shape.setType(torch._C.ListType.ofInts())
+        constant_of_shape = g.op("ConstantOfShape", shape, value_t=torch.tensor([2.0]))
+        self.run_test(g, constant_of_shape.node(), expect_tensor('Float', (1, 2, 3, 4)))
 
+    def test_constant_of_shape_dynamic(self):
+        # Test ConstantOfShape with input of prim::ListConstruct of dynamic tensor
+        rank = 4
+        g = self.create_empty_graph()
+        inputs = [g.addInput() for i in range(rank)]
+        shape = g.op("prim::ListConstruct", *inputs)
+        shape.setType(torch._C.ListType.ofInts())
+        constant_of_shape = g.op("ConstantOfShape", shape, value_t=torch.tensor([2.0]))
+        self.run_test(g, constant_of_shape.node(), expect_tensor('Float', dynamic_shape=(None, None, None, None)))
```

Comment:
Please add tests about reshape dim = 0 cases mentioned in https://github.com/pytorch/pytorch/pull/56053

Reply:
thx